src/error.js

```diff
@@ -539,6 +541,18 @@ export function detectNonAmpJs(win) {
   return false;
 }
 
+/**
+ * Returns true if the AMP runtime support for shadow DOM is used.
+ * @param {!Window} win
+ * @return {boolean}
+ * @visibleForTesting
+ */
+export function detectAmpShadowJs(win) {
+  const root = Services.ampdocServiceFor(win).getAmpDoc().getHeadNode();
+  const querySelector = 'script[src="' + urls.cdn + '/shadow-v0.js"]';
+  return !!root.querySelector(querySelector);
```

Comment:
Did we test that this works? I don't think it will for a few reasons -- one is the fact that the PWA shell that would include `shadow-v0.js`, not the AmpDoc's "head" node which is really the shadow root. 

What I think we want is changing `detectNonAmpJs()` to check the head node instead of `window.document`.